server/src/middleware/mod.rs

```diff
@@ -2,3 +2,5 @@
 
 /// Proxy `GET /path` to internal RPC methods.
 pub mod proxy_get_request;
+/// Proxy switch requset
```

Comment:
I think `request_middleware` is too vague and might confuse users.

Perhaps something like a `proxy_http_request` and write some better documentation that it 
exposes the HTTP request and it's possible to override the default jsonrpsee handler response.